File: src/main/java/no/nav/permitteringsskjemaapi/altinn/OrganisasjonerController.java
package no.nav.permitteringsskjemaapi.altinn;

import java.util.List;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.AllArgsConstructor;
import no.nav.security.token.support.core.api.Protected;

@RestController
@RequestMapping("/organisasjoner")
@AllArgsConstructor
@Protected
public class OrganisasjonerController {

    private final AltinnService altinnService;

    @GetMapping
    public List<AltinnOrganisasjon> hentOrganisasjoner() {
        return altinnService.hentOrganisasjoner();
    }
}

File: src/main/kotlin/no/nav/permitteringsskjemaapi/altinn/OrganisasjonerController.kt
package no.nav.permitteringsskjemaapi.altinn

import no.nav.permitteringsskjemaapi.config.INNSYN_ALLE_PERMITTERINGSSKJEMA
import no.nav.security.token.support.core.api.Protected
import org.springframework.web.bind.annotation.*

@RestController
@Protected
class OrganisasjonerController(
    private val altinnService: AltinnService
) {

    @GetMapping("/organisasjoner-v2")
    fun hentOrganisasjonerV2() = altinnService.hentAltinnTilganger(INNSYN_ALLE_PERMITTERINGSSKJEMA).hierarki
}